Changelog.txt

```diff
@@ -3429,9 +3429,219 @@ Additionally, the problem of zig-zag issue following in the South direction has
+	This trigger nows trigger before @Damage but after character @Hit trigger.
+	Args are same as the character @Hit trigger, but I and ARGO switch the values.
+	Args:
+	- I: The weapon being used.
+	- SRC: The character being hit.
+	- ARGO: The character doing the hitting.
+	- ARGN1: The amount of damage being dealt.
+	- ARGN2: The type(s) of damage.
+	- LOCAL.ARROW: UID of the arrow used (Archery only) [Read Only]
+	- LOCAL.ARROWHANDLED: If set true (1), clears the ammo stack variable.
+	- LOCAL.ITEMDAMAGECHANCE: Sets the damage chance for the weapon. (Default: 25%)
+	- LOCAL.ITEMPOISONREDUCTIONCHANCE: Sets the poison reduction chance for poisoned weapons.
+	- LOCAL.ITEMPOISONREDUCTIONAMOUNT: Sets the amount of poison that reduced from morez value.
+	Returns:
+	1: Cancels the combat swing of the character.
+- Added: Trigger @GetHit is added as an item trigger.
+	This trigger nows trigger before @Damage but after character @GetHit trigger.
+	Args are same as the character @GetHit trigger, but I changed to the item, and the character saves on ARGO.
+	Also LOCAL.ITEMDAMAGELAYER only works under character @GetHit trigger, it is only readable under item trigger.
+	Args:
+	- I: The clothing part take a hit.
+	- SRC: The character responsible for the damage.
+	- ARGO: The character being damaged.
+	- ARGN1: The amount of damage being applied. (Before slayer calculation if applicable)
+	- ARGN2: The type(s) of damage being applied.
+	- ARGN3: Hits will be fixed to this amount after the hit if the value is greaten than it (default: 0)
+	- LOCAL.ITEMDAMAGECHANCE: Sets the damage chance for the hitted part of the body. (Armor or shield, if any) (Default: 25%)
+	- LOCAL.DAMAGEPERCENTPHYSICAL: 	The amount in % of the physical damage received, only set if Elemental Engine is enabled. [Read Only]
+	- LOCAL.DAMAGEPERCENTPHYSICAL: The amount in % of the physical damage received, only set if Elemental Engine is enabled. [Read Only]
+	- LOCAL.DAMAGEPERCENTFIRE: The amount in % of the fire damage received, only set if Elemental Engine is enabled. [Read Only]
+	- LOCAL.DAMAGEPERCENTCOLD: The amount in % of the cold damage received, only set if Elemental Engine is enabled. [Read Only]
+	- LOCAL.DAMAGEPERCENTPOISON: The amount in % of the poison damage received, only set if Elemental Engine is enabled. [Read Only]
+	- LOCAL.DAMAGEPERCENTENERGY: The amount in % of the energy damage received, only set if Elemental Engine is enabled. [Read Only]
+	- LOCAL.ITEMDAMAGELAYER: The layer of the item being damaged. [Read Only]
+	Returns:
+	1: Prevents damage from being applied to the character.
+- Added: MAPVIEWRADAR, MAPVIEWSIZE and MAPVIEWSIZEMAX to sphere.ini that overrides default visual range values. (Issue: #1144)
+	Defaults:
+	  MapViewRadar=31
+	  MapViewSize=18
+	  MapViewSizeMax=24
+- Added: 'H' shortcut for variables to get the value as hexadecimal.
+	For example, if you use .tag.test 100
+	<dTAG.TEST> returns 100
+	<hTAG.TEST> returns 064
```

Comment:
It was like that, but it was changed in 56x, I don't know who did this change, but we are not using D keyword anymore because all variables return decimal in default, so I am using tons of hvals for my custom flag systems, honestly for me it is more important than D keyword at the moment.

Reply:
That's a problem then. Can you give me some snippets returning decimal where in the past it returned hex?

Reply:
On script side, everything was return in hex before. Now on X everything is default return in decimal.
Example, if you set tag.test=12
If you do .show tag.test you were seing  0C  on 55i and 56d.  If you want decimal you don.show dtag.test.
On X you always see default decimal.  There no way to see it in Hex. Htag is a good idea

Reply:
> That's a problem then. Can you give me some snippets returning decimal where in the past it returned hex?

For example;

```
[FUNCTION htest]
LOCAL.A = 010
TAG.A = 010
SERV.LOG A: <LOCAL.A> <TAG.A>
LOCAL.B = 100
TAG.B = 100
SERV.LOG B: <LOCAL.B> <TAG.B>
```
A returns as 16, and B as 100.

Reply:
Okay I changed it back to 16, for hex, and revert back the H shortcut update fully.

Reply:
So this means we have to go through all conditions where we check for number in tag and change it to dtag? 😫

Reply:
> So this means we have to go through all conditions where we check for number in tag and change it to dtag? 😫

Sorry for the inconvenience, but this should have worked as hex from the beginning, it was changed due to a simple mistake, the reason was unknown and we didn't touch it because we weren't sure if it was done on purpose or not.